src/ir/iteration.h

```diff
@@ -82,6 +82,30 @@ class ChildIterator {
   Iterator end() const { return Iterator(*this, children.size()); }
 };
 
+// Returns true if the current expression contains a certain kind of expression,
+// within the given depth of BFS. If depth is -1, this searches all children.
+template<typename T> bool containsChild(Expression* parent, int depth = -1) {
```

Comment:
When the depth is not limited, I think it would be more efficient to not use the ChildIterator, and do a normal traversal. `FindAll` gets a list of all the nodes, so maybe using that, or something similar to that but without saving the list?

(But this looks useful for the case where the depth is limited, which we'll probably need eventually, so I'm not suggesting we remove it.)

This can perhaps be moved to `ir/find_all.h` as it seems closely related?

Reply:
Done